aptos-move/block-executor/src/types.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright © Aptos Foundation
+
+use aptos_types::transaction::BlockExecutableTransaction as Transaction;
+use std::{collections::HashSet, fmt};
+
+#[derive(Eq, Hash, PartialEq, Debug)]
+pub enum InputOutputKey<K, T, I> {
+    Resource(K),
+    Group(K, T),
+    DelayedField(I),
+}
+
+pub struct ReadWriteSummary<T: Transaction> {
+    pub reads: HashSet<InputOutputKey<T::Key, T::Tag, T::Identifier>>,
```

Comment:
it's used in the construction. I kinda did this because both fields have the same signature, and so with "new()" function, order could be mistakenly swapped, while explicit constructor requires you to specify each.

but I'll change to private with new, as it is constructed only in two places anyways